gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemJavaStorageClientIntegrationTest.java

```diff
@@ -0,0 +1,47 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystemOptions.ClientType;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GoogleHadoopFileSystemJavaStorageClientIntegrationTest
+    extends GoogleHadoopFileSystemIntegrationTest {
+
+  @Before
+  public void before() throws Exception {
+    storageClientType = ClientType.STORAGE_CLIENT;
```

Comment:
having a mutable field is an anti pattern. A template method might be more appropriate here.

Reply:
It's a temporary fix to have a way to override storageClient. We didn't need it and will mostly clear it up once we have a fix for it.